### ISSUE TITLE
Set::fromArrays crash with zero arguments

### ISSUE THREAD

**Issue by Norbyte** [facebook/hhvm#6373]
\HH\Set::fromArrays() crashes HHVM when called with zero arguments and JIT is enabled.
The crash is not triggered with ```-vEval.Jit=0```.

HipHop VM 3.8.2-dev (dbg)
Compiler: heads/HHVM-3.8-0-ga83673ff227d68ca0253f6da131eea8cd3e82fc9
Repo schema: 9b7af269a1ce8c51980bc5c93dfc835424f02a9a

Repro:
```
<?php

\HH\Set::fromArrays();
```

`hhvm -vEval.Jit=1 crash.php`

Stack trace:
```
#0  get (this=0x0) at hhvm/hphp/runtime/base/smart-ptr.h:160
#1  get (this=0x0) at hhvm/hphp/runtime/base/type-array.h:77
#2  HPHP::ArrayIter::ArrayIter (this=0x7fffffffcba0, array=...) at hhvm/hphp/runtime/base/array-iterator.cpp:61
#3  0x00000000021952b8 in php_fromArrays<HPHP::c_Set> (_argv=..., _argc=<optimized out>) at hhvm/hphp/runtime/ext/ext_collections.cpp:4118
#4  HPHP::c_Set::ti_fromarrays (_argc=<optimized out>, _argv=...) at hhvm/hphp/runtime/ext/ext_collections.cpp:4401
#5  0x0000000008e00080 in ?? ()
#6  0x00007fffda47ffc0 in ?? ()
#7  0x000000000ce00708 in ?? ()
#8  0x0000000002e35e49 in enterTCHelper () at hhvm/hphp/runtime/vm/jit/translator-asm-helpers.S:66
#9  0x0000000001988a4e in HPHP::jit::x64::(anonymous namespace)::BackEnd::enterTCHelper (this=<optimized out>, start=<optimized out>, stashedAR=<optimized out>) at hhvm/hphp/runtime/vm/jit/back-end-x64.cpp:109
#10 0x0000000001ec3b35 in HPHP::jit::MCGenerator::enterTC (this=0x7fffe8fa0b00, start=0xce00700 "H\211\357\350\353\241\325\364\377\340\017\v\017\v\314\314H\211\357UH\211\345\377w\b\377\067H\307", <incomplete sequence \306>, stashedAR=stashedAR@entry=0x0)
    at hhvm/hphp/runtime/vm/jit/mc-generator.cpp:1094
#11 0x000000000183a0b3 in HPHP::jit::MCGenerator::enterTCAfterPrologue (this=<optimized out>, start=<optimized out>) at hhvm/hphp/runtime/vm/jit/mc-generator.h:225
#12 0x000000000187efcb in HPHP::enterVMAtFunc (enterFnAr=0x7fffda47ffc0, stk=<optimized out>, varEnv=<optimized out>) at hhvm/hphp/runtime/vm/bytecode.cpp:2027
#13 0x0000000002e3502c in operator() (__closure=<synthetic pointer>) at hhvm/hphp/runtime/vm/bytecode.cpp:2199
#14 exception_handler<HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, const HPHP::Func*, const HPHP::Variant&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags)::__lambda119> (action=...)
    at hhvm/hphp/runtime/vm/unwind-inl.h:30
#15 operator() (__closure=<synthetic pointer>) at hhvm/hphp/runtime/vm/bytecode.cpp:2073
#16 enterVMCustomHandler<HPHP::enterVM(HPHP::ActRec*, Action) [with Action = HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, const HPHP::Func*, const HPHP::Variant&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags)::__lambda119]::__lambda117> (action=..., ar=<optimized out>) at hhvm/hphp/runtime/vm/bytecode.cpp:2064
#17 enterVM<HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, const HPHP::Func*, const HPHP::Variant&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags)::__lambda119> (action=..., ar=<optimized out>)
    at hhvm/hphp/runtime/vm/bytecode.cpp:2073
#18 HPHP::ExecutionContext::invokeFunc (this=this@entry=0x7fffe5806370, retptr=retptr@entry=0x7fffffffcfb0, f=<optimized out>, args_=..., this_=this_@entry=0x0, cls=cls@entry=0x0, varEnv=0x7fffe5809b40, invName=invName@entry=0x0, 
    flags=flags@entry=HPHP::ExecutionContext::InvokePseudoMain) at hhvm/hphp/runtime/vm/bytecode.cpp:2200
#19 0x0000000002e6af12 in HPHP::ExecutionContext::invokeUnit (this=0x7fffe5806370, retval=retval@entry=0x7fffffffcfb0, unit=0x7fffe7ff45c0) at hhvm/hphp/runtime/vm/bytecode.cpp:2370
#20 0x00000000014d4c5a in HPHP::invoke_file_impl (res=..., path=..., once=once@entry=true, currentDir=currentDir@entry=0x2e86c53 "") at hhvm/hphp/runtime/base/builtin-functions.cpp:799
#21 0x0000000002e6ae11 in HPHP::invoke_file (s=..., once=once@entry=true, currentDir=currentDir@entry=0x2e86c53 "") at hhvm/hphp/runtime/base/builtin-functions.cpp:812
#22 0x00000000014dde7e in HPHP::include_impl_invoke (file=..., once=once@entry=true, currentDir=currentDir@entry=0x2e86c53 "") at hhvm/hphp/runtime/base/builtin-functions.cpp:838
#23 0x00000000015a2230 in HPHP::hphp_invoke (context=0x7fffe5806370, cmd="crash.php", func=func@entry=false, funcParams=..., funcRet=..., reqInitFunc="", reqInitDoc="", error=@0x7fffffffd1ff: false, errorMsg="", once=once@entry=true, warmupOnly=warmupOnly@entry=false, 
    richErrorMsg=richErrorMsg@entry=false) at hhvm/hphp/runtime/base/program-functions.cpp:1977
#24 0x00000000015a24ff in HPHP::hphp_invoke_simple (filename="crash.php", warmupOnly=warmupOnly@entry=false) at hhvm/hphp/runtime/base/program-functions.cpp:1938
#25 0x00000000015ad3b6 in HPHP::execute_program_impl (argc=argc@entry=5, argv=argv@entry=0x7fffffffe068) at hhvm/hphp/runtime/base/program-functions.cpp:1666
#26 0x00000000015ae14c in HPHP::execute_program (argc=argc@entry=5, argv=argv@entry=0x7fffffffe068) at hhvm/hphp/runtime/base/program-functions.cpp:1021
#27 0x000000000145539b in main (argc=5, argv=0x7fffffffe068) at hhvm/hphp/hhvm/main.cpp:60
```

**Comment by simonwelsh** [comment]
I can't replicate this with 3.9:

```
root@4:~# cat > crash.php
<?php

\HH\Set::fromArrays();
root@4:~# hhvm --version
HipHop VM 3.9.1 (rel)
Compiler: tags/HHVM-3.9.1-0-g0f72cfc2f0a01fdfeb72fbcfeb247b72998a66db
Repo schema: 6416960c150a4a4726fda74b659105ded8819d9c
root@4:~# hhvm -vEval.Jit=1 crash.php 
root@4:~# echo $?
0
```

**Comment by fredemmott** [comment]
@Norbyte : 3.8 is unsupported; can you reproduce this on a more recent version?

**Comment by Norbyte** [comment]
This seems to be fixed in v3.10
